texar/modules/encoders/xlnet_encoder.py

```diff
@@ -115,26 +101,7 @@ def __init__(self,
             torch.Tensor(1, 1, self._hparams.hidden_dim))
 
         if init:
-            if self.pretrained_model_dir:
-                init_xlnet_checkpoint(self, self.pretrained_model_dir)
-            elif self._hparams.initializer:
```

Comment:
Does our new design cover such logic? The user does not want to load the pre-trained checkpoint and want to initialize the model by specifying the `initializer` in `hparams`.

Reply:
No. As of now model loading is still somewhat coupled with specifying the architecture. That said, I don't think the original logic supports such functionality, as `initializer` is only checked when `self.pretrained_model_dir` is `None`.